File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # for i in range(len(nums)):
        #     for j in range(i+1,len(nums)):
        #         if nums[i] + nums[j] == target:
        #             return [i,j]
        
        hashmap = {}
        for idx, num in enumerate(nums):
            if (target - num) in hashmap:
                return [hashmap[target - num],idx]
            hashmap[num] = idx

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> store = new HashMap<Integer, Integer>();

        for(int idx = 0; idx < nums.length; idx++){
            if(store.containsKey((target - nums[idx]))){
                return new int[] {store.get(target - nums[idx]), idx};
            }
            store.put(nums[idx], idx);
        }
        return new int[]{};
    }
}